[REF] product_available_by_warehouse: remove domain route in show message

suniagajose opened the pull request:
## Summary

When it was used 'show message' field to see product available the domain route should consider only the routes that are available for sale order line.

To solve #1054 
## TODO
- [x] avoid route_ids domain in 'show message' onchange.
- [x] use sudo in warehouse search to show always the same info no matter what permission has the user.
- [x] swap positions between show_message field and product_id field

suniagajose commented:
Hi @JulioSerna 

Can you review this, please?

nhomar commented on product_available_by_warehouse/models/sale.py:
```diff
@@ -62,7 +62,8 @@ def onchange_show_message(self, show_message, product_id):
-            domain['route_id'] = [('id', 'in', route_ids)]
+            # TODO: uncomment until figure out a better way
+            # domain['route_id'] = [('id', 'in', route_ids)]
             res.update({'domain': domain})
```
sorry but I do not understand this.

You are passing a blank domain and wnat's the intention of this part.

I expect a unittest here if not we will be doing error-test until every fail generate hours of support.

Please create a unittest that given the border case return the quantity expected./

JulioSerna replied:
@nhomar 

this code do not passing blank domain, in fact is passing the exactly same domain returned by the original onchange

this change is because before it was modifying in wrong way and that is the reason of the change and  `TODO`  

[related issue](https://github.com/Vauxoo/addons-vauxoo/issues/1054)

nhomar commented on product_available_by_warehouse/views/sale_view.xml:
```diff
@@ -7,7 +7,7 @@
         <field name="inherit_id" ref="sale_stock.view_order_form_inherit" />
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="before">
+            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="after">
```
why are you changing the view? at least this is not mandatory in the requirement described in the task/issue.

JulioSerna replied:
@nhomar 
in fact, it is requirement for the customer, in usability is better to choose first the product
![produtc_after](https://cloud.githubusercontent.com/assets/7595505/18501578/393abfc6-7a17-11e6-8dad-efaca7c25ac9.png)
 than this
![produt_before](https://cloud.githubusercontent.com/assets/7595505/18501595/5279363e-7a17-11e6-9aa4-18229491baf7.png)

Commit:
[REF] product_available_by_warehouse: remove domain route in show message

When it was used 'show message' field to see product available the
domain route should consider only the routes that are available for
sale order line.

Also, swapped positions between show_message field and product_id
field. For usability is better to choose first the product.

In order to solve issue #1054

JulioSerna commented:
LGTM
👍
